jiant/proj/main/modeling/taskmodels.py

```diff
@@ -276,14 +282,94 @@ def get_output_from_encoder(encoder, input_ids, segment_ids, input_mask) -> Enco
         EncoderOutput containing pooled and unpooled model outputs as well as any other outputs.
 
     """
+    model_arch = ModelArchitectures.from_encoder(encoder)
+    if model_arch in [
+        ModelArchitectures.BERT,
+        ModelArchitectures.ROBERTA,
+        ModelArchitectures.ALBERT,
+        ModelArchitectures.XLM_ROBERTA,
```

Comment:
use one of`is_transformers_model_arch()` from model_resolution.py style methods?

Reply:
Not sure what you mean here - `is_transformers_model_arch()` checks if it is one of the Transformer-based models (which is all of them, currently), but this has different behavior for different architectures.

Reply:
I am trying to look for opportunities to reduce the custom if-else statements for each model architecture for maintenance and ease of adding a new model. In an ideal world, we should be able to get the outputs without checking the model architecture